activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -136,9 +142,17 @@ def as_json(options = nil) # :nodoc:
   end
 end
 
+ActiveSupport::KERNEL_RESPOND_TO = Kernel.instance_method(:respond_to?)
```

Comment:
I wasn't sure the best place to put this (either in terms of namespace or location in the code). feedback welcome.